src/Oscoin/Consensus/Mining.hs

```diff
@@ -34,20 +38,21 @@ mineBlock
        )
     => BlockStoreReader c tx s m
     -> Consensus c tx s m
-    -> Evaluator st tx b
+    -> Evaluator st tx o
     -> Timestamp
-    -> m (Maybe (Block c tx (Sealed c s), st, [Receipt c tx b]))
```

Comment:
Interesting -- this _used_ to have this type, but when we discussed this with @kim during my work on #402, we made the miner return also the state and the list of receipts. The rationale was that block evaluation/validation might still fail, and you might not want to update the state _just yet_. What do you think?

Reply:
I agree that `mineBlock` does all the wrong things here. However this is a stopgap measure to reduce the code in `Oscoin.Node.mineBlock` and still make the tests pass. I intend to move the interactions with the state/block/receipt store in `mineBlock` behind an abstraction. And the state and receipts will only be written after a mined block is accepted by the protocol.